### PR TITLE
ServiceBus may 2021 GA release

### PR DESCRIPTION
**Pull request by hemanttanwar**
Release preparation for version  ServiceBus 7.2.1

- Upgraded `azure-core` dependency to `1.16.0`.
- Upgraded `azure-core-amqp` dependency to `2.0.5`.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus-track2-perf/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Release History
 
-## 1.0.0-beta.1 (Unreleased)
+## 1.0.0-beta.1 (2021-05-12)
```
Comment (alzimmermsft) [inline sdk/servicebus/azure-messaging-servicebus-track2-perf/CHANGELOG.md:3]: nit: Not needed

File: sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
 # Release History
 
-## 7.3.0-beta.2 (Unreleased)
+## 7.2.1 (2021-05-12)
+### Fixed
+- Fix the issue where 'ServiceBusProcessorClient:maxConcurrentCalls' will result in more messages being cached in SDK
```
Comment (alzimmermsft) [inline sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:5]:
```suggestion
- Fixed an issue where `ServiceBusProcessorClient:maxConcurrentCalls` will result in more messages being cached in the SDK  
```

File: sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,13 @@
 # Release History
 
-## 7.3.0-beta.2 (Unreleased)
+## 7.2.1 (2021-05-12)
+### Fixed
+- Fix the issue where 'ServiceBusProcessorClient:maxConcurrentCalls' will result in more messages being cached in SDK  
+  and message lock expire.
```
Comment (alzimmermsft) [inline sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:6]: This doesn't really make sense with the following line, was the issue a case where messages would be cached even after the message lock expired?

**Review by alzimmermsft — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,13 @@
 # Release History
 
-## 7.3.0-beta.2 (Unreleased)
+## 7.2.1 (2021-05-12)
+### Fixed
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:4]: nit: `Fixes`

**Review by srnagar — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md (hunk 4 above)
Comment (hemanttanwar) [reply sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:4]: I am following this guidelines for the  changelog
https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#change-logs

**Pull request merged**